### PR TITLE
:memo: Data model ADR

### PR DESCRIPTION
**Pull request by evaline-ju**
For #13

### CODE REVIEW

**Review by markstur — COMMENTED** [review]
Just wondering... I tend to think of the data models as both input and/or output to Predict (I'm ignoring train for now).  I like using simpler types as input when possible.  So does it make sense to clarify if "data of concrete models" is meant to imply inference output and/or training input...etc?  It's a little vague to me right now.  Maybe the word "results" should be used more. Just curious if that fits this ADR.

**Comment by evaline-ju** [comment]
@markstur Thanks for the input. I've tried to rephrase to be a bit clearer. The wording was likely too vague in an attempt at generality, but concretely I was imagining we could use data models for any of inference input (say a raw document structure) or output (classification result) or training input (any sort of training record). I tend to associate the word “result” with “output,” which would be a bit constraining.

File: docs/adrs/010-data-model-definition.md [hunk 1]
```diff
@@ -20,5 +20,6 @@ Accepted
 
 - Caikit users will be able to access model information through data models
 - Caikit users may face one extra level of indirection if they want to access mathematical representations of their data models results (e.g. arrays).
-- Caikit will provide the same data model per task [TODO: link to task ADR]. 
+- Caikit will provide the same data model per AI task [TODO: link to task ADR].
 - Data models will be backed by a common backend [TODO: link to backends, protobuf ADRs].
+- Caikit model authors will have to translate model inputs and outputs to work with data models. This has implications on usability and stability.
```
Comment (markstur) [inline docs/adrs/010-data-model-definition.md:25]: I think s/Caikit model authors/Caikit module authors/

Comment (evaline-ju) [reply docs/adrs/010-data-model-definition.md]: I was trying to go for some consistency with the README personas provided here: https://github.com/caikit/caikit#user-profiles switched s/Caikit/AI/

Comment (hickeyma) [reply docs/adrs/010-data-model-definition.md]: I think model makes more sense as we are looking at the profile from the high level go AI models.

**Review by joerunde — APPROVED** [review]

**Pull request merged**